mod/uservalidationbyemail/actions/uservalidationbyemail/change_email.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Resends validation emails to a user and/or changes a newly registered users email address
+ */
+
+$user = elgg_call(ELGG_IGNORE_ACCESS | ELGG_SHOW_DISABLED_ENTITIES, function() {
+	return get_user((int) get_input('guid'));
+});
+
+if (!$user instanceof \ElggUser) {
+	return elgg_error_response(elgg_echo('error:missing_data'));
+}
+
+$change_secret = get_input('change_secret');
+
+$correct_secret = elgg_build_hmac([
+	$user->guid,
+	$user->time_created,
+])->getToken();
+
+if ($change_secret !== $correct_secret) {
+	return elgg_error_response(elgg_echo('actionunauthorized'));
+}
```

Comment:
the `Elgg\Security\HMac` from `elgg_build_hmac()` has a function `matchesToken()` to do the comparison